crates/hc/src/save.rs

```diff
@@ -96,3 +97,65 @@ pub fn list(hc_dir: PathBuf, verbose: usize) -> anyhow::Result<()> {
     msg!("{}", out);
     Ok(())
 }
+
+lazy_static::lazy_static! {
+    static ref FILE_LOCKS: tokio::sync::Mutex<Vec<usize>> = tokio::sync::Mutex::new(Vec::new());
+}
+
+/// Lock this setup as running live and advertise the port
+pub async fn save_port(mut hc_dir: PathBuf, path: &Path, port: u16) -> anyhow::Result<()> {
+    use std::io::Write;
+    std::fs::create_dir_all(&hc_dir)?;
+    let paths = load(hc_dir.clone())?;
+    let index = match paths.into_iter().enumerate().find(|p| p.1 == path) {
+        Some((i, _)) => i,
+        None => return Ok(()),
+    };
+    hc_dir.push(format!(".hc_live_{}", index));
+    match std::fs::OpenOptions::new()
+        .write(true)
+        .create_new(true)
+        .open(hc_dir)
+    {
+        Ok(mut file) => {
+            writeln!(file, "{}", port)?;
+            let mut lock = FILE_LOCKS.lock().await;
+            lock.push(index);
+        }
+        Err(e) => match e.kind() {
+            std::io::ErrorKind::AlreadyExists => {}
+            _ => return Err(e.into()),
+        },
+    }
+
+    Ok(())
+}
+
+/// Load setup ports
+pub fn load_ports(hc_dir: PathBuf) -> anyhow::Result<Vec<Option<u16>>> {
+    let mut ports = Vec::new();
+    let paths = load(hc_dir.clone())?;
+    for (i, _) in paths.into_iter().enumerate() {
+        let mut hc = hc_dir.clone();
+        hc.push(format!(".hc_live_{}", i));
+        if hc.exists() {
+            let live = std::fs::read_to_string(hc)?;
+            let p = live.lines().next().and_then(|l| l.parse::<u16>().ok());
+            ports.push(p)
+        }
+    }
+    Ok(ports)
+}
+
+/// Release lock on setups
+pub async fn release_ports(hc_dir: PathBuf) -> anyhow::Result<()> {
```

Comment:
Seems like the only fn in this file that is `async`, and on top of it it does blocking fs operations. Suspicius.

Reply:
Yep because I didn't use async file reading for this because the files are so small it would likely be slower